### DISCUSSION TITLE
What does "Loaded env from /var/task/.env.production" means ?

### DISCUSSION THREAD

**Discussion by ErnestBrandi** [vercel/vercel#7011] [General] [answered]
Hi,

checking my app logs on Logflare I noticed some of my requests have the "Loaded env from /var/task/.env.production" mention. Those requests usually take more time to be processed (+ 2s). 

My best guess about it is that those requests initiate a new lambda function therefore loading a full environment (cold boot) when the following re-use a lambda function (hot boot). Is that true ?

Thanks.

**Comment by leerob** [comment]
I'm guessing that is a build log, not a log from your lambda.

https://vercel.com/docs/concepts/deployments/logs#build-logs

**Comment by ErnestBrandi** [comment]
I am talking about API calls here, so I guess those are pre-built, aren't they ? I noticed this usually happens after a certain period of inactivity on the website though.
This the message I get
```
"message": "START RequestId: 5d808ed5-154c-4a8e-9d23-42fb1e618a14 Version: $LATEST\n2021-11-17T08:12:17.741Z\tundefined\tINFO\tinfo  - Loaded env from /var/task/.env.production\nEND RequestId: 5d808ed5-154c-4a8e-9d23-42fb1e618a14\nREPORT RequestId: 5d808ed5-154c-4a8e-9d23-42fb1e618a14\tDuration: 2044.47 ms\tBilled Duration: 2045 ms\tMemory Size: 1024 MB\tMax Memory Used: 127 MB\tInit Duration: 445.03 ms\t\n",
```
this is the same message for subsequent "normal" requests 
```
"message": "START RequestId: 65d42187-f0c2-4cd8-b434-a7af644b3f79 Version: $LATEST\nEND RequestId: 65d42187-f0c2-4cd8-b434-a7af644b3f79\nREPORT RequestId: 65d42187-f0c2-4cd8-b434-a7af644b3f79\tDuration: 195.14 ms\tBilled Duration: 196 ms\tMemory Size: 1024 MB\tMax Memory Used: 129 MB\t\n",
```

**Comment by saidhasyim** [comment]
Hey @ErnestBrandi , did you manage to solve it? I hit the same exact issue with API call.

**Comment by vincentwinkel** [comment]
Same here

**Comment by nikos-portmoto** [comment]
Same for me

**Comment by ordinz** [comment]
I'm getting `Loaded env from /var/task/.env`

**Comment by Fasa2509** [comment]
Same here, my app was working fine two weeks ago and now it just broke. These api calls use a lot of memory and take too much time to complete, any solution?

**Comment by davevilela** [comment]
Any updates?

**Comment by octavieploye** [comment]
same here :(

**Comment by eslam980** [comment]
Still no update ?

**Comment by collasanta** [comment]
?

**Comment by almaron** [comment]
Getting the same error. Was looking forward to moving to a pro account, but the projects aren't even loading

**Comment by PhYdrogen** [comment]
I had this issue i don't know if i could help but, i had to add my .env.local in my next.conf.file like this [medium]( https://aude53.medium.com/set-environment-variables-with-next-js-and-vercel-e544c0460a48)

**Comment by lucvega** [comment]
Same issue here

**Comment by TooTallNate** [comment, marked as answer]
`Loaded env from /var/task/.env.production` means that the environment variables from the `.env.production` file in your project have been loaded into the process. This would typically happen at boot-up.

The reason that these requests take longer is because it is a cold boot of the serverless function, as evidenced by the `Init Duration: 445.03 ms` in the logs that @ErnestBrandi shared.

So there is no issue here, things are working as expected. The correlation between the loading of env vars message and the slow request are symptoms of the cold boot.
